client/src/app/worker/autoupdate-stream.ts

```diff
@@ -0,0 +1,269 @@
+import * as fzstd from 'fzstd';
+
+import {
+    ErrorDescription,
+    ErrorType,
+    isCommunicationError,
+    isCommunicationErrorWrapper
+} from '../gateways/http-stream/stream-utils';
+import { joinTypedArrays, splitTypedArray } from '../infrastructure/utils/functions';
+import { AutoupdateSubscription } from './autoupdate-subscription';
+import { AutoupdateSetEndpointParams } from './interfaces-autoupdate';
+
+export class AutoupdateStream {
+    public failedCounter: number = 0;
+
+    private abortCtrl: AbortController = undefined;
+    private activeSubscriptions: AutoupdateSubscription[] = null;
+    private active: boolean = false;
+    private error: any | ErrorDescription = null;
+    private restart: boolean = false;
+
+    public get subscriptions(): AutoupdateSubscription[] {
+        return this._subscriptions;
+    }
+
+    public get failedConnects(): number {
+        return this.failedCounter;
+    }
+
+    constructor(
+        private _subscriptions: AutoupdateSubscription[],
+        public queryParams: string,
+        private endpoint: AutoupdateSetEndpointParams,
+        private authToken: string
+    ) {}
+
+    /**
+     * Clones this stream with the specified subscriptions
+     *
+     * @param subscriptions The subscriptions handled by the created stream
+     */
+    public cloneWithSubscriptions(subscriptions: AutoupdateSubscription[]): AutoupdateStream {
+        return new AutoupdateStream(subscriptions, this.queryParams, this.endpoint, this.authToken);
+    }
+
+    /**
+     * Closes the stream
+     */
+    public abort(): void {
+        if (this.abortCtrl !== undefined) {
+            // @ts-ignore
+            this.abortCtrl.abort();
+        }
+    }
+
+    /**
+     * Opens a new connection to autoupdate.
+     * Also this function registers this stream inside all subscriptions
+     * handled by this stream.
+     *
+     * resolves when fetch connection is closed
+     */
+    public async start(
+        force?: boolean
+    ): Promise<{ stopReason: 'error' | 'aborted' | 'unused' | 'resolved' | 'in-use'; error?: any }> {
```

Comment:
`'error' | 'aborted' | 'unused' | 'resolved' | 'in-use'` looks like you may turn it into a type or an enum or smth